Sources/Core/ObjectiveCIR.swift

```diff
@@ -246,11 +246,26 @@ public struct ObjCIR {
     }
 
     static func enumStmt(_ enumName: String, body: () -> [String]) -> String {
+        let enums = body()
+        let enumType: String
+        switch enums.count {
+        case 0 ... Int(UInt8.max):
+            enumType = "unsigned char"
+        case Int(UInt8.max) ... Int(UInt16.max):
+            enumType = "unsigned short"
```

Comment:
These new cases are handling the positive (> = 0) ranges. Would there be a benefit to also covering the negative (unsigned) parts of these ranges, too? As written, an enum that has the values (-1, 0, 1) will be generated using the `NSInteger` type.

Reply:
I agree; this definitely needs to be expanded upon. The example you gave would be a `char` enum. This is a WIP, mostly so I can get tips on the best way to add this to the tool. I'm new to its design pattern, and I need some tips. Thanks!

Reply:
I think the approach you're taking is good. This switch can be expanded to cover those additional ranges, and I think we'll end up with a pretty good result.

Reply:
Please see the updated PR.